core/includes/database/select.inc

```diff
@@ -488,29 +508,38 @@ interface SelectQueryInterface extends QueryConditionInterface, QueryAlterableIn
    * Helper function to build most common HAVING conditional clauses.
    *
    * This method can take a variable number of parameters. If called with two
-   * parameters, they are taken as $field and $value with $operator having a value
-   * of IN if $value is an array and = otherwise.
+   * parameters, they are taken as $field and $value with $operator having a
+   * value of IN if $value is an array and = otherwise.
    *
-   * @param $field
+   * @param string $field
    *   The name of the field to check. If you would like to add a more complex
    *   condition involving operators or functions, use having().
-   * @param $value
-   *   The value to test the field against. In most cases, this is a scalar. For more
-   *   complex options, it is an array. The meaning of each element in the array is
-   *   dependent on the $operator.
-   * @param $operator
-   *   The comparison operator, such as =, <, or >=. It also accepts more complex
-   *   options such as IN, LIKE, or BETWEEN. Defaults to IN if $value is an array
-   *   = otherwise.
+   * @param mixed $value
+   *   The value to test the field against. In most cases, this is a scalar. For
+   *   more complex options, it is an array. The meaning of each element in the
+   *   array is dependent on the $operator.
+   * @param string $operator
+   *   The comparison operator, such as =, <, or >=. It also accepts more
+   *   complex options such as IN, LIKE, or BETWEEN. Defaults to IN if $value is
+   *   an array, and = otherwise.
+   *
    * @return QueryConditionInterface
    *   The called object.
    */
   public function havingCondition($field, $value = NULL, $operator = NULL);
 
+  /**
+   * String magic method.
+   *
+   * @return string
+   *   The prepared SQL statement as a string.
+   */
+  public function __toString();
```

Comment:
It's OK to add this to interface SelectQueryInterface now, right? It's just for documentation purposes... Or, could this have any side effects?

Reply:
By adding it to the interface, it means that classes that implement this interface **must** provide a `__toString()` method, otherwise it will cause an error. However, since `__toString()` is crucial for a query to work at all, we can be confident that any implementing classes already have `__toString()`.

But also, yes it's added primarily so we have a place to put the documentation.